src/idl_parser.cpp

```diff
@@ -853,17 +853,40 @@ void Parser::SerializeStruct(const StructDef &struct_def, const Value &val) {
 
 CheckedError Parser::ParseTable(const StructDef &struct_def, std::string *value,
                                 uoffset_t *ovalue) {
-  EXPECT('{');
+  // We allow tables both as JSON object{ .. } with field names 
+  // or vector[..] with all fields in order
+  
+  const bool is_nested_list = Is('['); 
+
+  if (is_nested_list) {
+    NEXT();
+  } else {
+    EXPECT('{');
+  }
+
   size_t fieldn = 0;
   for (;;) {
-    if ((!opts.strict_json || !fieldn) && Is('}')) { NEXT(); break; }
-    std::string name = attribute_;
-    if (Is(kTokenStringConstant)) {
-      NEXT();
+    if ((!opts.strict_json || !fieldn) && Is(is_nested_list ? ']' : '}')) { NEXT(); break; }
+
```

Comment:
Ok done (sorry was misled by other parts of that file which have more blanklines to think it was acceptable, like Parser::SkipJsonObject)